llvm/lib/Target/AMDGPU/AMDGPUAtomicOptimizer.cpp

```diff
@@ -402,34 +413,30 @@ Value *AMDGPUAtomicOptimizerImpl::buildReduction(IRBuilder<> &B,
 
   // Reduce within each pair of rows (i.e. 32 lanes).
   assert(ST->hasPermLaneX16());
-  V = B.CreateBitCast(V, IntNTy);
```

Comment:
Please submit an NFC cleanup patch that just removes unnecessary bitcasting, before adding support for new atomic operations.